sdk/metric/aggregator/maxsumcount/msc.go

```diff
@@ -65,15 +73,21 @@ func (c *Aggregator) Count() (int64, error) {
 }
 
 // Max returns the maximum value in the checkpoint.
+// The error value aggregator.ErrEmptyDataSet
+// will be returned if (due to a race condition) the checkpoint was
+// computed before the first value was set.
 func (c *Aggregator) Max() (core.Number, error) {
+	if c.checkpoint == unsetMaxSumCount(c.kind) {
```

Comment:
Because the MaxSumCount aggregator does not take a lock during Update, each field of the state will be independently modified, and a race with Update could happen such that the count and sum are non-zero but the max field hasn't been set. So, I think you should test `c.checkpoint.max == c.kind.Minimum()` here.

(Note: this means, now, that the minimum and maximum values are considered invalid. This might be worth adding comments about, i.e., that if you record a measure value which is the Minimum possible value, that the Max() function will return a no-data error.  I don't think this will be a problem, but it would probably _become_ a problem if this library supported unsigned integer instruments.  Then a value of Record(0) would lead to a no-data error in Max(). If we did ever support unsigned instruments, I we'd probably want to offset by one or something. Nevermind.)

Reply:
Related note: https://gitter.im/open-telemetry/opentelemetry-go?at=5dd7871489fce12f25a5a76c